src/proteus/dswx_hls.py

```diff
@@ -1169,73 +1176,76 @@ def _apply_aerosol_class_remapping_wtr1_class(wtr_1_layer,
             Remapped WTR-1 class
     """
 
-    to_mask_array = None
+    to_remap_array = None
     for fmask_value in fmask_values:
-        if to_mask_array is None:
-            to_mask_array = fmask == fmask_value
+        if to_remap_array is None:
+            to_remap_array = fmask == fmask_value
         else:
-            to_mask_array |= fmask == fmask_value
-    to_mask_array &= wtr_1_layer == input_wtr1_class
-    wtr_1_layer[to_mask_array] = output_wtr1_class
+            to_remap_array |= fmask == fmask_value
```

Comment:
This can probably be simplified with `numpy.isin` ([link](https://numpy.org/doc/stable/reference/generated/numpy.isin.html#numpy.isin)).

Reply:
Thank you, @nemo794 ! I didn't know that function, it definitely simplifies the logic